packages/venia-concept/src/components/Checkout/__tests__/braintreeDropin.spec.js

```diff
@@ -27,3 +57,57 @@ test('creates an instance of the underlying dropin on mount', async () => {
         expect(dropin.create).toHaveBeenCalledTimes(1);
     });
 });
+
+test('returns paymentNonce', async () => {
+    const component = TestRenderer.create(<BraintreeDropin {...props} />);
+    const nonce = 'imanonce';
+    component.root.children[0].instance.dropinInstance = {
+        requestPaymentMethod: jest.fn().mockImplementation(() => nonce)
+    };
+    await component.root.children[0].instance.requestPaymentNonce();
+
+    expect(mockSuccess).toHaveBeenCalledWith(nonce);
+});
+
+test('returns stored payment info on failure to get payment nonce', async () => {
+    const component = TestRenderer.create(<BraintreeDropin {...props} />);
+    component.root.children[0].instance.dropinInstance = {
+        requestPaymentMethod: jest.fn().mockImplementation(() => {
+            throw new Error();
+        })
+    };
+    await component.root.children[0].instance.requestPaymentNonce();
+
+    expect(mockSuccess).toHaveBeenCalledWith('storedData');
+});
+
+test('throws an error on failure to get payment nonce and no stored payment info', async () => {
+    const component = TestRenderer.create(<BraintreeDropin {...props} />);
+    component.root.children[0].instance.dropinInstance = {
+        requestPaymentMethod: jest.fn().mockImplementation(() => {
+            throw new Error();
+        })
+    };
+    await component.root.children[0].instance.requestPaymentNonce();
+
+    expect(mockError).toHaveBeenCalled();
+});
+
+test('requests the payment nonce if instructed by parent', () => {
+    const component = TestRenderer.create(
+        <BraintreeDropin
+            isRequestingPaymentNonce={true}
+            onError={mockError}
+            onSuccess={mockSuccess}
+        />
+    );
+
+    const mockRequestPaymentNonce = jest.fn();
+    component.root.children[0].instance.requestPaymentNonce = mockRequestPaymentNonce;
+    component.root.children[0].instance.dropinInstance = true;
+    component.root.children[0].instance.componentDidUpdate({
+        isRequestingPaymentNonce: false
+    });
```

Comment:
In a test titled "requests the payment nonce if instructed by parent", it's weird to see `isRequestingPaymentNonce` set to `false` here, but this is the `prevProps`, right?

Perhaps creating an object called `prevProps` and passing that in would make it easier to read / not weird.

Reply:
I'm going to leave this just cause I expect people familiar with React to know that this is `prevProps`.